diff --git a/configure b/configure
index da631a45e..f4d450cfe 100755
--- a/configure
+++ b/configure
@@ -103,6 +103,7 @@ all_platforms="${all_platforms} arm64-darwin20-gcc"
 all_platforms="${all_platforms} arm64-linux-gcc"
 all_platforms="${all_platforms} arm64-win64-gcc"
 all_platforms="${all_platforms} arm64-win64-vs15"
+all_platforms="${all_platforms} arm64-win64-vs16"
 all_platforms="${all_platforms} armv7-android-gcc"   #neon Cortex-A8
 all_platforms="${all_platforms} armv7-darwin-gcc"    #neon Cortex-A8
 all_platforms="${all_platforms} armv7-linux-rvct"    #neon Cortex-A8
@@ -270,7 +271,6 @@ ARCH_EXT_LIST="
 HAVE_LIST="
     ${ARCH_EXT_LIST}
     vpx_ports
-    pthread_h
     unistd_h
 "
 EXPERIMENT_LIST="
